sklearn/linear_model/_ridge.py

```diff
@@ -2038,9 +2033,36 @@ def fit(self, X, y, sample_weight=None):
         the validation score.
         """
         cv = self.cv
+
+        check_scalar_alpha = partial(
+            check_scalar,
+            target_type=numbers.Real,
+            min_val=0.0,
+            include_boundaries="neither",
+        )
+
+        if isinstance(self.alphas, (np.ndarray, list, tuple)):
+            n_alphas = 1 if np.ndim(self.alphas) == 0 else len(self.alphas)
+            if n_alphas != 1:
+                for index, alpha in enumerate(self.alphas):
+                    alpha = check_scalar_alpha(alpha, f"alphas[{index}]")
+            elif np.ndim(self.alphas) == 0:
+                # converting a single scalar into a np.ndarray
+                # (e.g. `x=np.array(1)`)
+                # creates a 0d-array whose value can be accessed with
+                # `x[()]`
+                self.alphas[()] = check_scalar_alpha(self.alphas[()], "alphas")
```

Comment:
Uhm it seems that we don't have any test for this case.
Could we add such use case by passing a 0d array